### PR TITLE
Tab hide disable

### LINKED ISSUE #3088: Add the ability to hide and disable tabs
This one comes up from time to time and we should probably add it. This will affect both `Tabs` and `TabbedContent`. The ability to either:

- Disable a tab (so it's still visible but the tab shows disabled and, if you can select it at all, the content (in the case of `TabbedContent`) of the pane is all disabled).
- Hide a tab (and so it can't be seen at all, but can be made visible again; in the case of `TabbedContent` the related pane would be invisible and the widgets within it unreachable).

### PR DESCRIPTION
**Pull request by rodrigogiraoserrao**
This fixes #3088.

Adds API to enable/disable/show/hide tabs.
 - A `Tab` widget can be disabled directly via the reactive `.disabled`, which then fires a `Tab.Disabled` message, which is turned into a `Tabs.Disabled` message.
 - Preferably, and in line with the show/hide API, one can use the methods on `Tabs` or `TabbedContent`.

### CODE REVIEW

**Comment by willmcgugan** [comment]
Can I see a screenshot / video of a disabled tab?

**Review by willmcgugan — CHANGES_REQUESTED** [review]
Nice work. Just one request.

**Comment by rodrigogiraoserrao** [comment]
Screenshots from `textual run tests/snapshot_tests/snapshot_apps/modified_tabs.py`.

Tabs 1 and 2 are disabled, tab 1 was disabled while active:

Untitled1

Tab 5 active while 1 and 2 are disabled:

Untitled 2

File: src/textual/widgets/_tabs.py [hunk 1]
```diff
@@ -104,17 +105,35 @@ class Tab(Static):
...
         """A tab was clicked."""
 
         tab: Tab
         """The tab that was clicked."""
 
-        def __init__(self, tab: Tab) -> None:
-            self.tab = tab
-            super().__init__()
+    @dataclass
+    class Disabled(Message):
```
[the remark this answers is not in the document]
Comment (rodrigogiraoserrao) [reply src/textual/widgets/_tabs.py:125]: That'd be right. But `Tab.Disabled` + `Tab.Enabled` are kind of internal, caught by `Tabs`, and re-posted as `Tabs.TabDisabled` and `Tabs.TabEnabled`, mimicking the way `Tab.Clicked` & `Tabs.TabActivated` work.

Do you think it's worth adding the controls to `Tab.Disabled` and `Tab.Enabled`?
(Do you want me to do the same for `Tab.Clicked`..?)

[the remark this answers is not in the document]
Comment (willmcgugan) [reply src/textual/widgets/_tabs.py:125]: Only the messages visible to the dev need it.

**Review by willmcgugan — APPROVED** [review]

**Pull request merged**